### ISSUE TITLE
Resize quality

### ISSUE THREAD

**Issue by achirkof** [ccorcos/meteor-clientside-image-manipulation#11]
Hello.
Great package. I tried this and it's very simple and useful. But if I compare quality between processImage with quality 1.0 and server side resizing using GraphicsMagick with quality 100%.
![compare](http://i.imgur.com/1rMXoVE.png)

Same image size but left image quality poor.
Is it possible to increase quality of resized image?

**Comment by ccorcos** [comment]
Ah. Its possible, but thats on you for PR ;) GraphicsMagick is doing some nice anti-aliasing and smoothing while this one is just applying it to a canvas and then ripping the image off

https://github.com/ccorcos/meteor-clientside-image-manipulation/blob/master/lib/processImage.coffee#L69-L82

perhaps theres something built into canvas for this. I'm not sure...

**Comment by achirkof** [comment]
I found example how to make image smoothing [here](http://stackoverflow.com/a/17862644/1709413). But I can't modify package. Can you add some anti-aliasing to resized image?

**Comment by ccorcos** [comment]
Oh interesting -- thats pretty easy. You can fork and submit a PR. Otherwise I'll get around to this eventually. Not at home personal computer right now...

**Comment by achirkof** [comment]
Yes, it should be easy for you but for me... I'm know nothing about canvas and it will be very hard task. I will thank you if you're doing this.

**Comment by ccorcos** [comment]
Basically, right [here](https://github.com/ccorcos/meteor-clientside-image-manipulation/blob/master/lib/processImage.coffee#L86) you'll just want to recursively resize the image by no more than half at each step until you eventually get to the desired size...

**Comment by ccorcos** [comment]
Just want to be honest with you here -- I'm not using this package at all right now. I'm actually in the process of not using Meteor at all anymore. So if you want this feature, you'll likely have to implement it yurself.

**Comment by achirkof** [comment]
Ok. I have to find solution.

**Comment by ccorcos** [comment]
Ok. I did a little work for you -- you'll need to test it out though.

```
downsize = (img, [targetWidth, targetHeight]) ->
  if img.width/targetWidth > 2 or img.height/targetHeight > 2
    canvas = document.createElement('canvas')
    ctx = canvas.getContext("2d")
    ctx.drawImage img, 0, 0, img.width/2, img.height/2
    newImg = ctx.getImageData(0, 0, img.width/2, img.height/2)
    return downsize(newImg, [targetWidth, targetHeight])
  else
    canvas = document.createElement('canvas')
    ctx = canvas.getContext("2d")
    ctx.drawImage img, 0, 0, img.width/2, img.height/2
    return newImg = ctx.getImageData(0, 0, targetWidth, targetHeight)
```

Basically just recursively downsize ensuring that you never downsize by more than 1/2 each step. 

To develop, just do this:

```
git clone https://github.com/ccorcos/meteor-clientside-image-manipulation.git ccorcos:clientside-image-manipulation
cd ccorcos:clientside-image-manipulation
cd example
meteor
```

Edit the code and you should be good to go testing with the example app.

**Comment by achirkof** [comment]
So... I tried to put this part of code into code after [this](https://github.com/ccorcos/meteor-clientside-image-manipulation/blob/master/lib/processImage.coffee#L86) line,
comment [this](https://github.com/ccorcos/meteor-clientside-image-manipulation/blob/master/lib/processImage.coffee#L56-L64) part and change it with your code. But no results. Quality stil the same.
To be honets I don't understand coffee.

**Comment by ccorcos** [comment]
Hmm. Coffeescript translated directly to JS. Use this to gain and understanding:

http://js2.coffee/
